src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,49 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (!animal.hidden && !(animal instanceof Carnivore)) {
```

Comment:
```suggestion
  bite(animal) {
    if (!animal.hidden && animal instanceof Herbivore) {
```
There is no need to overcomplicate the condition